llvm/include/llvm/ProfileData/MemProf.h

```diff
@@ -535,7 +519,7 @@ class RecordLookupTrait {
 
   data_type ReadData(uint64_t K, const unsigned char *D,
                      offset_type /*Unused*/) {
-    Record = IndexedMemProfRecord::deserialize(Schema, D);
+    Record = IndexedMemProfRecord::deserialize(Schema, D, Version1);
```

Comment:
Suggest having a CurrentVersion variable in the header where the versions are defined and set it to Version1 and use here and elsewhere. Will greatly simplify updating to the new version.

Reply:
When I teach the reader/writer to use `Version2`, I'm planning to take a version as a constructor parameter:

```
 RecordLookupTrait(IndexedVersion Version, const MemProfSchema &S)
     : Version(Version), Schema(S) {}
```

and pass `Version` to `IndexedMemProfRecord::deserialize` here.  Is that OK?

Reply:
Maybe just put part of that in this patch, since it would be confined to this same file - i.e. have a member of RecordLookupTrait that you initialize to Version1 and pass down via deserialize. Ditto I guess for RecordWriterTrait and serialize()/serializedSize().

Reply:
I've included the changes to `RecordLookupTrait` and `RecordWriterTrait` in the latest iteration.